### ISSUE TITLE
French localisation

### ISSUE THREAD

**Issue by pylapp** [mlemgroup/mlem#1124]
## Introduction

Today the app provides only english texts. All the strings are hardcoded and not extracted in localizables files.
I propose ot bring localizaiton for both english and french by:
1. extracting all hard-coded strings to a localizable strings dict
2. add french translations

## Requirements

- Extract all hard-coded values
- Define localizable string dict
- Add french translations

## Discussion

Could be quite easy, I can do it in one evening.

## What I Need

Nothing particular, just get your agreement before starting dev, it can induce conflcits between branches so I need to ship this evolution fast.

WDYT @buresdv @EricBAndrews?

**Comment by Sjmarf** [comment]
Hey there! Full localisation is something we'd love to support at some stage.

At the moment, we're busy rewriting our entire codebase on the `dev2` branch. The next major update will be Mlem v2.0, which will use this new codebase. Mlem on the `dev2` branch is only partially implemented, and many UI components are missing. You're welcome to localise what we've got so far, if that appeals to you. Alternatively, I can let you know when Mlem v2.0 is finished so that you can add the French translations. I don't think it would be worth adding translations to the v1 codebase on the `dev` branch at this stage, as it will be replaced with `dev2` in the next month or two.

**Comment by pylapp** [comment]
Sounds interesting, maybe it could be more efficient indeed if the v2 code base is ready ; waiting for such version will be better because we will be able to add French translations in one time for all the views.

Feel free to ping me when you think the code base can be pulled so as to get my French wording 😁

**Comment by Sjmarf** [comment]
Sure, I'll let you know when it's ready then :D

**Comment by pylapp** [comment]
Nice 👌

**Comment by pylapp** [comment]
Do you think the codebase is ready? ;)

**Comment by Sjmarf** [comment]
Hey! The v2 codebase contains all of the core features at this point, but we're still working on adding moderation features back in. 

If you would like, you're welcome to localize what we've got so far - we've set up a `Localizable` file. Because we haven't finished certain features yet, you'd have to come back and localize those features closer to the final Mlem v2 release.

Alternatively, we can let you know when we're approaching the v2 release so that you can do them all at once, which you might prefer.

**Comment by pylapp** [comment]
Ok so maybe it will be more efficient if I wait for a finalized v2 version to not forget any wording :)

**Comment by Sjmarf** [comment]
Yeah that's more efficient :) I'll let you know when we're close to releasing.

**Comment by EricBAndrews** [comment]
@pylapp the v2 codebase has no further planned wording changes, so you're clear to do your thing

**Comment by pylapp** [comment]
@EricBAndrews Okay! I should work on that next week :)

**Comment by EricBAndrews** [comment]
Please note that #1796 introduces one additional string to localize ("Load More")

**Comment by pylapp** [comment]
@EricBAndrews I plan to submit a PR at the end of this week :)

**Comment by pylapp** [comment]
@EricBAndrews FYI PR https://github.com/mlemgroup/mlem/pull/1803 is ready for review :)

**Comment by Sjmarf** [comment]
Hey @pylapp! We'll be releasing an update to Mlem in a couple of weeks. The update contains some new text that could be translated into french. We won't be making any changes to the text between now and when the update is released, so you're welcome to translate the new strings if you'd like to :)

**Comment by pylapp** [comment]
For sure! I can check that next week 😄

**Comment by pylapp** [comment]
@Sjmarf You can get the updated french wording with https://github.com/mlemgroup/mlem/pull/1954
